lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -279,6 +289,25 @@ func AddInvoice(ctx context.Context, cfg *AddInvoiceConfig,
 				continue
 			}
 
+			// If the user has manually defined private hop hints, ignore
+			// any channels not in the array.
+			if len(invoice.PrivateHops) > 0 {
+				match := false
+				for _, privateChan := range invoice.PrivateHops {
```

Comment:
This may be better to implement using a `map[uint64]struct{}`

Reply:
Agree, done